staging/src/k8s.io/apiserver/pkg/server/filters/wrap.go

```diff
@@ -21,14 +21,15 @@ import (
 
 	"k8s.io/klog/v2"
 
+	"k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apiserver/pkg/server/httplog"
 )
 
 // WithPanicRecovery wraps an http Handler to recover and log panics (except in the special case of http.ErrAbortHandler panics, which suppress logging).
 func WithPanicRecovery(handler http.Handler) http.Handler {
 	return withPanicRecovery(handler, func(w http.ResponseWriter, req *http.Request, err interface{}) {
-		if err == http.ErrAbortHandler {
+		if net.IsAbortHandlerError(err) {
```

Comment:
maybe we should at least log why we dropped the request?

Reply:
```
if errors.Is(http.ErrAbortHandler) {
  err := recover()
  klog.Error(err)
  panic(http.ErrAbortHandler)
}
```

Reply:
from golang docs `To abort a handler so the client sees an interrupted response but the server doesn't log an error, panic with the value ErrAbortHandler`

maybe we should actually throw `ErrAbortHandler` instead of `errConnKilled` ?
that would be seen as `  timeout_test.go:220: Get "http://127.0.0.1:61805": EOF` on the client side.